File: app/src/main/java/net/veldor/flibustaloader/database/dao/BooksDownloadScheduleDao.java
package net.veldor.flibustaloader.database.dao;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;

import net.veldor.flibustaloader.database.entity.BooksDownloadSchedule;
import net.veldor.flibustaloader.database.entity.ReadedBooks;

import java.util.List;

@Dao
public interface BooksDownloadScheduleDao {
    @Query("SELECT COUNT(id) FROM BooksDownloadSchedule")
    int getQueueSize();

    @Query("SELECT * FROM BooksDownloadSchedule LIMIT 1")
    BooksDownloadSchedule getFirstQueuedBook();

    @Query("SELECT * FROM BooksDownloadSchedule WHERE bookId = :id")
    BooksDownloadSchedule getBookById(String id);


    @Query("SELECT * FROM BooksDownloadSchedule")
    List<BooksDownloadSchedule> getAllBooks();


    @Query("SELECT * FROM BooksDownloadSchedule")
    LiveData<List<BooksDownloadSchedule>> getAllBooksLive();

    @Insert
    void insert(BooksDownloadSchedule book);

/*    @Update
    void update(ReadedBooks book);
*/
    @Delete
    void delete(BooksDownloadSchedule book);
}

File: app/src/main/java/net/veldor/flibustaloader/database/dao/BooksDownloadScheduleDao.kt
package net.veldor.flibustaloader.database.dao

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.Query
import net.veldor.flibustaloader.database.entity.BooksDownloadSchedule

@Dao
interface BooksDownloadScheduleDao {
    @get:Query("SELECT COUNT(id) FROM BooksDownloadSchedule")
    val queueSize: Int

    @get:Query("SELECT * FROM BooksDownloadSchedule LIMIT 1")
    val firstQueuedBook: BooksDownloadSchedule?

    @Query("SELECT * FROM BooksDownloadSchedule WHERE bookId = :id")
    fun getBookById(id: String?): BooksDownloadSchedule?

    @get:Query("SELECT * FROM BooksDownloadSchedule")
    val allBooks: List<BooksDownloadSchedule>?

    @get:Query("SELECT * FROM BooksDownloadSchedule")
    val allBooksLive: LiveData<List<BooksDownloadSchedule>>?

    @Insert
    fun insert(book: BooksDownloadSchedule?)

    @Delete
    fun delete(book: BooksDownloadSchedule?)
}